process/block/preprocess/transactions.go

```diff
@@ -86,22 +92,29 @@ func NewTransactionPreprocessor(
 	if check.IfNil(economicsFee) {
 		return nil, process.ErrNilEconomicsFeeHandler
 	}
-	if check.IfNil(miniBlocksCompacter) {
-		return nil, process.ErrNilMiniBlocksCompacter
-	}
 	if check.IfNil(gasHandler) {
 		return nil, process.ErrNilGasHandler
 	}
 	if check.IfNil(blockTracker) {
 		return nil, process.ErrNilBlockTracker
 	}
+	if check.IfNil(addressConverter) {
+		return nil, process.ErrNilAddressConverter
+	}
+
+	//TODO maybe inject this and check if this is needed to be used also for smartContractResults
```

Comment:
you could create at coordinator level and push for all preprocessors and postprocessors.

Reply:
the number of smart contracts is always much smaller than transactions so the size of the blockbody should be fine.

Reply:
Done